packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -213,41 +212,24 @@ export default function useOnBlockDrop( targetRootClientId, targetBlockIndex ) {
 	} = useDispatch( 'core/block-editor' );
 
 	return {
```

Comment:
The object is spread out after the hook is called and passed on to `useDropZone`, which in turns adds and removes the drop zones from the provider if any of these callback change. `useMemo` on the object won't make a difference here. I removed `useCallback` here because it was masking the infinite look issue. Everything should work properly without the `useCallback`, just like it did before the refactoring.